taichi/ir/type.cpp

```diff
@@ -105,17 +104,32 @@ CustomIntType::CustomIntType(int num_bits,
                              bool is_signed,
                              Type *compute_type,
                              Type *physical_type)
-    : compute_type(compute_type),
-      physical_type(physical_type),
+    : compute_type_(compute_type),
+      physical_type_(physical_type),
       num_bits_(num_bits),
       is_signed_(is_signed) {
   if (compute_type == nullptr) {
     auto type_id = is_signed ? PrimitiveTypeID::i32 : PrimitiveTypeID::u32;
-    this->compute_type =
+    this->compute_type_ =
         TypeFactory::get_instance().get_primitive_type(type_id);
   }
 }
 
+CustomFloatType::CustomFloatType(Type *digits_type,
+                                 Type *compute_type,
+                                 float64 scale)
+    : digits_type_(digits_type), compute_type_(compute_type), scale_(scale) {
+  TI_ASSERT(digits_type->is<CustomIntType>());
+  TI_ASSERT(digits_type->as<CustomIntType>()->get_is_signed());
```

Comment:
Is this assertion necessary? GlobalLoadStmt checks both possibilities and it makes sense for the digits type to be customized depending on applications

Reply:
Good point. Let me remove this assertion :-) Thanks for pointing this out.